### PR TITLE
[Routing] Allow using services in the route condition

### LINKED ISSUE #16714: [Routing] Allow using services in the route condition
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#44405
| PR author(s) | @renanbr
| Merged in    | 6.1

### PR DESCRIPTION
**Pull request by renanbr**
Documents https://github.com/symfony/symfony/pull/44405

Closes https://github.com/symfony/symfony-docs/issues/16714

### CODE REVIEW

File: routing.rst [hunk 1]
```diff
@@ -462,11 +462,19 @@ and can use any of these variables created by Symfony:
...
 
-You can also use this function:
+You can also use these functions:
 
 ``env(string $name)``
     Returns the value of a variable using :doc:`Environment Variable Processors <configuration/env_var_processors>`
 
+``service(string $alias)``
+    Returns a routing condition service.
+    You'll have to add the ``#[AsRoutingConditionService]`` attribute or ``routing.condition_service`` tag to your service if you want to use it in the condition.
```
Comment (wouterj) [inline routing.rst]:
```suggestion
    You'll have to add the ``#[AsRoutingConditionService]`` attribute or ``routing.condition_service``
    tag to your service if you want to use it in the condition.
```

(we always add a line break after the first word that crosses the 72nd column)

File: routing.rst [hunk 2]
```diff
@@ -462,11 +462,19 @@ and can use any of these variables created by Symfony:
...
+
+.. versionadded:: 6.1
+
+    The ``service(string $alias)`` function was introduced in Symfony 6.1.
```
Comment (wouterj) [inline routing.rst]:
```suggestion
    The ``service(string $alias)`` function and ``#[AsRoutingConditionService]`` attribute
    was introduced in Symfony 6.1.
```

**Review by wouterj — APPROVED** [review]
Thank you!

**Comment by javiereguiluz** [comment]
Renan, thanks a lot for this contribution!

**Pull request merged**